napari/components/_interaction_box_mouse_bindings.py

```diff
@@ -46,7 +50,107 @@ def __init__(self, viewer, interaction_box_visual):
+            hasattr(self._viewer.layers.selection.active, 'mode')
+            and self._viewer.layers.selection.active.mode == 'transform'
+            and self._viewer.dims.ndisplay > 2
+        ):
+            self._viewer.layers.selection.active.mode = 'pan_zoom'
+
+    def _on_dim_change(self, event):
+        """Gets called when changing order of dims to make sure interaction box is using right extent and transform"""
+        viewer = self._viewer
+        if (
+            hasattr(viewer.layers.selection.active, 'mode')
+            and viewer.layers.selection.active.mode == 'transform'
+        ):
+            viewer.overlays.interaction_box.points = (
+                viewer.layers.selection.active.extent.data[
+                    :, list(viewer.dims.displayed)
+                ]
+                - 0.5
+            )
+            viewer.overlays.interaction_box.transform = (
+                viewer.layers.selection.active.affine.set_slice(
+                    list(viewer.dims.displayed)
+                )
+            )
+
+    def _on_mode_change(self, event):
+        """Gets called on mode change to enable interaction box in transform mode"""
+        viewer = self._viewer
+        if event.mode == 'transform':
+            if viewer.dims.ndisplay > 2:
+                warnings.warn(
+                    trans._(
+                        'Interactive transforms in 3D are not yet supported.',
+                        deferred=True,
+                    ),
+                    category=UserWarning,
+                )
+                viewer.layers.selection.active.mode = 'pan_zoom'
+                return
+            viewer.layers.selection.active = event.source
+            # The -0.5 is nescessary because the pixel at (0,0) actually extends to (-0.5,0.5) (in case of the image layer)
+            viewer.overlays.interaction_box.points = (
+                event.source.extent.data[:, list(viewer.dims.displayed)] - 0.5
+            )
+            viewer.overlays.interaction_box.transform = (
+                event.source.affine.set_slice(list(viewer.dims.displayed))
+            )
+
+            viewer.overlays.interaction_box.show = True
+            viewer.overlays.interaction_box.show_vertices = True
+            viewer.overlays.interaction_box.show_handle = True
+            viewer.overlays.interaction_box.allow_new_selection = False
+
+            viewer.overlays.interaction_box.events.transform_drag.connect(
+                self._on_transform_change
+            )
+        else:
+            viewer.overlays.interaction_box.show = False
+            viewer.overlays.interaction_box.points = None
+            viewer.overlays.interaction_box.transform = Affine()
+
+    def _on_transform_change(self, event):
+        """Gets called when the interaction box is transformed to update transform of the layer"""
+
+        # TODO: right now the transform in the not displayed dimension is reset, probably would be better to preserve it

```

Comment:
Agreed that this is important. I think it needs to be addressed to take this from a WIP to something that is mergeable, unless we continue to hide/flag-guard this new behavior.

Reply:
>     * I assume we want the interaction box to update when `Layer.affine` is modified. That doesn't seem to happen yet. Is that as simple as connecting an event to make the interaction box overlay update? Or are there some unknowns/risks?

No, should be as simple as that. In my mind, the transform property of the interaction box and the affine property of the layers should be linked, i. e. the controller listens for changes in both and then updates the other one.



>     * The graphical design choices are non-trivial, so I think we should either find a way to get this in without graphically exposing the functionality, or get a little design feedback (e.g. in a community meeting).

Maybe the way to go is to remove the UI elements for now so that the transform mode can only be accessed by setting `.mode` of the layer to `'transform'`. I'll be happy to demo this in a community meeting. Actually there are some other elements I would like to demo that I think are super important to make this an efficient registration system.

Reply:
This has been addressed by using `replace_slice` right? If so, remove the TODO. If not, I think it's fine to leave this and merge without the fix because the new behavior is fairly well hidden behind the keyboard shortcut at the moment.